pkg/statistics/handle/autoanalyze/priorityqueue/calculator.go

```diff
@@ -14,23 +14,55 @@
 
 package priorityqueue
 
-// WeightCalculator is an interface for calculating weights of analysis jobs.
-type WeightCalculator interface {
-	CalculateWeight(job *TableAnalysisJob) float64
-}
+import "math"
+
+const (
+	// EventNone represents no special event.
+	eventNone = 0.0
+	// EventNewIndex represents a special event for newly added indexes.
+	eventNewIndex = 2.0
+)
+
+// TODO: make these configurable.
+const (
+	changeRatioWeight = 0.6
+	sizeWeight        = 0.1
+	analysisInterval  = 0.3
+)
 
 // PriorityCalculator implements the WeightCalculator interface.
-type PriorityCalculator struct {
-	threshold float64
-}
+type PriorityCalculator struct{}
```

Comment:
I want to move the formula out because I believe we can improve/redesign it in the future. For instance, if we decide to try a different formula in the future, we won't need to update the job's interface. We can simply implement another calculator.